packages/wonder-blocks-announcer/src/index.ts

```diff
@@ -0,0 +1,334 @@
+ * @param {string} message The message to send.
+ * @param {PolitenessLevel} level Polite or assertive announcements
+ * @param {number} removalDelay Optional duration to remove a message after sending. Defaults to 5000ms.
+ * @returns {string} IDREF for targeted live region element
+ */
+export function sendMessage({
+    message,
+    level = "polite", // TODO: decide whether to allow other roles, i.e. role=`timer`
+    removalDelay,
+}: SendMessageProps): string {
+    announcer = Announcer.getInstance();
+
+    if (typeof jest === "undefined") {
+        setTimeout(() => {
+            return announcer?.announce(message, level, removalDelay);
+        }, 100);
+    } else {
+        // If we are in a test environment, announce without waiting
+        return announcer.announce(message, level, removalDelay);
+    }
+    return "";
+}
+/**
+ * Public API method to clear screen reader messages after sending.
+ * Clears all regions by default.
+ * @param {string} id Optional id of live region element to clear.
+ */
+export function clearMessages(id?: string) {
+    if (id && document?.getElementById(id)) {
+        announcer?.clear(id);
+    } else if (document) {
+        announcer?.clear();
+    }
+}
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
+    private static _instance: Announcer;
+    node: HTMLElement | null = null;
+    regionFactory: RegionFactory = {
+        count: 2,
+        aIndex: 0,
+        pIndex: 0,
+    };
+    dictionary: RegionDictionary = new Map();
+
+    private constructor() {
+        if (typeof document !== "undefined") {
+            const topLevelId = `wbAnnounce`;
+            // Check if our top level element already exists
+            const announcerCheck = document.getElementById(topLevelId);
+
+            // Init new structure if the coast is clear
+            if (announcerCheck === null) {
+                this.init(topLevelId);
+            }
+            // The structure exists but references are lost, so help HMR recover
+            else {
+                this.rebootForHMR();
+            }
+        }
+    }
+    /**
+     * Singleton handler to ensure we only have one Announcer instance
+     * @returns {Announcer}
+     */
+    static getInstance() {
+        if (!Announcer._instance) {
+            Announcer._instance = new Announcer();
+        }
+        return Announcer._instance;
+    }
+    /**
+     * Internal initializer method to create live region elements
+     * Prepends regions to document body
+     * @param {string} id ID of the top level node (wbAnnounce)
+     */
+    init(id: string) {
+        this.node = document.createElement("div");
+        this.node.id = id;
+        this.node.setAttribute("data-testid", id);
+
+        Object.assign(this.node.style, srOnly);
+
+        const aWrapper = this.createRegionWrapper("assertive");
+        this.createDuplicateRegions(aWrapper, "assertive");
```

Comment:
**question:** Could you please elaborate why we need to create duplicate regions here? I was looking at the story in the browser and I'm not sure what should expect from that.

Reply:
It's an architectural decision to try and prevent messages from being dropped by AT, something I picked up back in the Angular.js days. Coupled with the debounce logic I'm working on, I think I'll keep it in for now to ensure unique messages are appended and always picked up by AT.

Reply:
ah ok, that makes sense thanks! Do you think you could add a short comment explaining this context? so other folks could understand the why of this approach. I'm sure my future self will appreciate that :).

Reply:
+1 me (and future me) too! Especially around alternating index. It would be great to document why we do this and how the live regions are structured!

Reply:
Good call. I added some comments to this and the types!